src/Factory.php

```diff
@@ -432,12 +438,16 @@ private function isPersisting(): bool
         }
 
         try {
-            self::configuration()->objectManagerFor($this->class);
-
-            return true;
+            $classMetadata = self::configuration()->objectManagerFor($this->class)->getClassMetadata($this->class);
         } catch (\RuntimeException $e) {
             // entity not managed (perhaps Embeddable)
             return false;
         }
+
+        if ($classMetadata instanceof ODMClassMetadata && $classMetadata->isEmbeddedDocument) {
+            return false;
+        }
+
+        return true;
```

Comment:
```suggestion
        return $classMetadata instanceof ODMClassMetadata && $classMetadata->isEmbeddedDocument;
```

Reply:
This would return `true` when it should return `false`. I think the fact that I had to really look hard to figure this out means we should leave it verbose.

Reply:
Then, it could be 
```suggestion
        return !($classMetadata instanceof ODMClassMetadata && $classMetadata->isEmbeddedDocument);
```

But it doesn't really matter :-) matter of taste